src/plugins/visualizations/public/wizard/group_selection/group_selection.tsx

```diff
@@ -0,0 +1,266 @@
+                      visType={visType}
+                      key={visType.name}
+                      onVisTypeSelected={props.onVisTypeSelected}
+                      showExperimental={props.showExperimental}
+                    />
+                  ))}
+                </div>
+              </EuiFlexItem>
+            )}
+          </EuiFlexGrid>
+          <EuiSpacer size="m" />
+          <EuiDescriptionList
+            className="visNewVisDialogGroupSelection__footerDescriptionList"
+            type="responsiveColumn"
+          >
+            <EuiDescriptionListTitle className="visNewVisDialogGroupSelection__footerDescriptionListTitle">
+              <FormattedMessage
+                id="visualizations.newVisWizard.learnMoreText"
+                defaultMessage="Want to learn more?"
+              />
+            </EuiDescriptionListTitle>
+            <EuiDescriptionListDescription>
+              <EuiLink href={visualizeGuideLink} target="_blank" external>
+                <FormattedMessage
+                  id="visualizations.newVisWizard.readDocumentationLink"
+                  defaultMessage="Read documentation"
+                />
+              </EuiLink>
+            </EuiDescriptionListDescription>
+          </EuiDescriptionList>
+        </div>
+      </EuiModalBody>
+    </>
+  );
+}
+
+const VisGroup = ({ visType, onVisTypeSelected }: VisCardProps) => {
+  const onClick = () => onVisTypeSelected(visType);
+  return (
+    <EuiFlexItem>
+      <EuiCard
+        titleSize="xs"
+        title={<span data-test-subj="visTypeTitle">{visType.title}</span>}
+        onClick={onClick}
+        isDisabled={isVisTypeAlias(visType) && visType.disabled}
+        betaBadgeLabel={
+          isVisTypeAlias(visType) && visType.disabled
+            ? i18n.translate('visualizations.newVisWizard.basicTitle', {
+                defaultMessage: 'Basic',
+              })
+            : undefined
+        }
+        betaBadgeTooltipContent={
+          isVisTypeAlias(visType) && visType.disabled
+            ? i18n.translate('visualizations.newVisWizard.basicLicenseRequired', {
+                defaultMessage: 'This feature requires a Basic License',
+              })
+            : undefined
+        }
+        data-test-subj={`visType-${visType.name}`}
+        data-vis-stage={!('aliasPath' in visType) ? visType.stage : 'alias'}
+        aria-label={`visType-${visType.name}`}
+        description={visType.description || ''}
+        layout="horizontal"
+        icon={<EuiIcon type={visType.icon || 'empty'} size="xl" color="secondary" />}
+        className="visNewVisDialog__groupsCard"
+      />
+    </EuiFlexItem>
+  );
+};
+
+const ToolsGroup = ({ visType, onVisTypeSelected, showExperimental }: VisCardProps) => {
+  // hide the experimental visualization if lab mode is not enabled
+  if (!showExperimental && visType.stage === 'experimental') {
+    return null;
+  }
+  const onClick = () => onVisTypeSelected(visType);
```

Comment:
nit: please review your PR and try to find places where we can wrap some callback's into `useCallback` hook. 

https://en.reactjs.org/docs/hooks-reference.html#usecallback